### PR TITLE
update eslint to last version

### PR DESCRIPTION
**Pull request by PinkaminaDianePie**
we used very old eslint config which blocked us from using latest eslint and flow. i replaced this config by one we use in main application, so from now on we should have same configs in main app and SG

### CODE REVIEW

File: flow-typed/npm/classnames_v2.x.x.js [hunk 1]
```diff
@@ -3,21 +3,21 @@
```
Comment (maciekbb) [inline flow-typed/npm/classnames_v2.x.x.js]: flow typed should be ignored

Comment (PinkaminaDianePie) [reply flow-typed/npm/classnames_v2.x.x.js]: done

File: src/components/flex/Flex.jsx [hunk 2]
```diff
@@ -202,7 +198,8 @@ export type FlexPropsType = {
    *            component content
    *          </Flex>
    */
-  marginLeft?: FlexMarginsType
+  marginLeft?: FlexMarginsType,
+  ...
```
Comment (vergilius) [inline src/components/flex/Flex.jsx]: what's that?

File: src/components/form-elements/Textarea.jsx [hunk 3]
```diff
@@ -3,22 +3,18 @@
...
   NORMAL: 'normal',
   TALL: 'tall',
-  XTALL: 'xtall'
+  XTALL: 'xtall',
 };
 
 export type TextareaPropsType = {
   // $FlowFixMe any generic prop types here broke autocomplete, so let's leave it as is for now
-  type?: string | any => React$Node,
+  type?: string | (any => React$Node),
```
Comment (vergilius) [inline src/components/form-elements/Textarea.jsx]: tricky one

File: src/components/list/MenuList.jsx [hunk 4]
```diff
@@ -1,37 +1,44 @@
...
+    {
+      [`sg-menu-list--${size}`]: size !== SIZE.NORMAL,
+    },
+    className
+  );
 
   return (
     <ul {...props} className={listClass}>
-      {items.map(({...elementProps}, index) =>
+      {items.map(({...elementProps}, index) => (
```
Comment (vergilius) [inline src/components/list/MenuList.jsx]: side note: does that make sense to do spread here? `{...elementProps}`

File: src/docs/components/ComponentSettings.jsx [hunk 5]
```diff
@@ -12,10 +13,10 @@ type PropsType = {
   settings: Array<{
     name: string,
     values: any,
-    required?: boolean
+    required?: boolean,
   }>,
   values: {[string]: number | string | boolean},
-  onChange: (string, number | string | boolean) => mixed
+  onChange: (string, number | string | boolean) => mixed,
```
Comment (vergilius) [inline src/docs/components/ComponentSettings.jsx]: side note: this looks suspicious (not related to PR) 🕵

File: src/docs/webpackEntry.jsx [hunk 6]
```diff
@@ -40,46 +43,25 @@ const BasicsPage = () => <GeneratePage padeIndex="0" />;
 const ComponentsPage = () => <GeneratePage padeIndex="1" />;
 const ContainersPage = () => <GeneratePage padeIndex="2" />;
 const InteractivePage = () => (
-  <React.Fragment>
+  <>
     <h1 className="main-header">Brainly style guide - {navigation[3].name}</h1>
     {sections}
-  </React.Fragment>
+  </>
```
Comment (vergilius) [inline src/docs/webpackEntry.jsx]: side note: hm, fragments break syntax highlighting (?) :(

File: src/components/flex/Flex.jsx (hunk 2 above)
Comment (vergilius) [reply src/components/flex/Flex.jsx]: @PinkaminaDianePie that's single thing I would check again ☝️

Comment (PinkaminaDianePie) [reply src/components/flex/Flex.jsx]: it should be in next pull request only, good catch ;)

**Pull request merged**